File: easy/q7 reverseInteger.cpp
/*

Given a 32-bit signed integer, reverse digits of an integer.

*/


class Solution {
public:
    int reverse(int x) {
        string s = to_string(x);
        string copy = s;
        int len = s.length(); 
        int start = 0;
        
        if (x < 0) {
            start++;
        }
        
        for (int a = start; a < len; a++) {
            if (x < 0) {
                copy[a] = s[len-a];
            }
            else {
                copy[a] = s[len-a-1];
            }
        }
        
        try {
            return stoi(copy);
        }
        catch (out_of_range &e) {
            return 0;
        }      
        
    }
};

File: easy/q7 reverseInteger.py
# Given a signed 32-bit integer x, return x with its digits reversed. If reversing x causes the value to go outside the signed 32-bit integer range [-231, 231 - 1], then return 0.

# Assume the environment does not allow you to store 64-bit integers (signed or unsigned).

class Solution:
    def reverse(self, x: int) -> int:
        newS = ""
        if x < 0:
            newS += '-'
            x = int(str(x)[1:])
        
        newS += str(x)[::-1]
        if int(newS) > (2**31)-1 or int(newS) < -2**31:
            return 0
        else:
            return int(newS)
        